contrib/python/podman/podman/libs/pods.py

```diff
@@ -0,0 +1,164 @@
+
+        with client() as podman:
+            self._refresh(podman)
+
+    def _refresh(self, podman):
+        pod = podman.GetPod(self._ident)
+        super().update(pod['pod'])
+
+        for k, v in self.data.items():
+            setattr(self, k, v)
+        return self
+
+    def inspect(self):
+        """Retrieve details about pod."""
+        with self._client() as podman:
+            results = podman.InspectPod(self._ident)
+        obj = json.loads(results['pod'], object_hook=fold_keys())
+        obj['id'] = obj['config']['id']
+        return collections.namedtuple('PodInspect', obj.keys())(**obj)
+
+    def kill(self, signal_=signal.SIGTERM, wait=25):
+        """Send signal to all containers in pod.
+
+        default signal is signal.SIGTERM.
+        wait n of seconds, 0 waits forever.
+        """
+        running = FoldedString(self.status)
+
+        with self._client() as podman:
+            podman.KillPod(self._ident, signal_)
+            timeout = time.time() + wait
+            while True:
+                # pylint: disable=maybe-no-member
+                self._refresh(podman)
+                if running != 'running':
+                    return self
+
+                if wait and timeout < time.time():
+                    raise TimeoutError()
+
+                time.sleep(0.5)
+        return self
+
+    def pause(self):
+        """Pause all containers in the pod."""
+        with self._client() as podman:
+            podman.PausePod(self._ident)
+            return self._refresh(podman)
+
+    def refresh(self):
+        """Refresh status fields for this pod."""
+        with self._client() as podman:
+            return self._refresh(podman)
+
+    def remove(self, force=False):
+        """Remove all containers in the pod, the pod, return pod ident.
+
+        force=True, stop any running container.
+        """
+        with self._client() as podman:
+            results = podman.RemovePod(self._ident, force)
+        return results['pod']
+
+    def restart(self):
+        """Restart all containers in the pod."""
+        with self._client() as podman:
+            podman.RestartPod(self._ident)
+            return self._refresh(podman)
+
+    def stats(self):
+        """Stats on all containers in the pod."""
+        with self._client() as podman:
+            results = podman.GetPodStats(self._ident)
+        for obj in results['containers']:
+            yield collections.namedtuple('ContainerStats', obj.keys())(**obj)
+
+    def start(self):
+        """Start all containers in the pod."""
+        with self._client() as podman:
+            podman.StartPod(self._ident)
+            return self._refresh(podman)
+
+    def stop(self):
+        """Stop all containers in the pod."""
+        with self._client() as podman:
+            podman.StopPod(self._ident)
+            return self._refresh(podman)
+
+    def top(self):
+        """Display stats for all containers exit."""
```

Comment:
drop "exit" here?

Reply:
I wonder if exit was meant for another window...  :) . Gone now.